src/tests/test_resultset.py

```diff
@@ -109,177 +110,478 @@ def test_invalid_operation_error(result_set, fname, parameters):
 
 def test_resultset_config_autolimit_dict(result, config):
     config.autolimit = 1
-    resultset = ResultSet(result, config).fetch_results()
+    resultset = ResultSet(result, config, statement=None, conn=Mock())
     assert resultset.dict() == {"x": (0,)}
 
 
-def test_resultset_with_non_sqlalchemy_results(config):
-    df = pd.DataFrame({"x": range(3)})  # noqa
-    conn = Connection(engine=create_engine("duckdb://"))
-    result = conn.execute("SELECT * FROM df")
-    assert ResultSet(result, config).fetch_results() == [(0,), (1,), (2,)]
+@pytest.fixture
+def results(ip_empty):
+    engine = create_engine("duckdb://")
+    session = engine.connect()
 
+    session.execute(text("CREATE TABLE a (x INT,);"))
 
-def test_none_pretty(config):
-    conn = Connection(engine=create_engine("sqlite://"))
-    result = conn.execute("create table some_table (name, age)")
-    result_set = ResultSet(result, config)
-    assert result_set.pretty is None
-    assert "" == str(result_set)
+    session.execute(text("INSERT INTO a(x) VALUES (1),(2),(3),(4),(5);"))
 
+    sql = "SELECT * FROM a"
+    results = session.execute(text(sql))
 
-def test_lazy_loading(result, config):
-    resultset = ResultSet(result, config)
-    assert len(resultset._results) == 0
-    resultset.fetch_results()
-    assert len(resultset._results) == 3
+    results.fetchmany = Mock(wraps=results.fetchmany)
+    results.fetchall = Mock(wraps=results.fetchall)
+    results.fetchone = Mock(wraps=results.fetchone)
+
+    yield results
+
+    session.close()
+
+
+@pytest.fixture
+def duckdb_sqlalchemy(ip_empty):
+    conn = Connection(create_engine("duckdb://"))
+    yield conn
+
+
+@pytest.fixture
+def sqlite_sqlalchemy(ip_empty):
+    conn = Connection(create_engine("sqlite://"))
+    yield conn
+
+
+@pytest.fixture
+def duckdb_dbapi():
+    conn_ = duckdb.connect(":memory:")
+    conn = DBAPIConnection(conn_)
+    yield conn
 
 
 @pytest.mark.parametrize(
-    "autolimit, expected",
+    "session, expected_value",
     [
-        (None, 3),
-        (False, 3),
-        (0, 3),
-        (1, 1),
-        (2, 2),
-        (3, 3),
-        (4, 3),
+        ("duckdb_sqlalchemy", {}),
+        ("duckdb_dbapi", {"Count": {}}),
+        ("sqlite_sqlalchemy", {}),
     ],
 )
-def test_lazy_loading_autolimit(result, config, autolimit, expected):
-    config.autolimit = autolimit
-    resultset = ResultSet(result, config)
-    assert len(resultset._results) == 0
-    resultset.fetch_results()
-    assert len(resultset._results) == expected
+def test_convert_to_dataframe_create_table(session, expected_value, request):
+    session = request.getfixturevalue(session)
+
+    mock = Mock()
+    mock.displaylimit = 100
+    mock.autolimit = 100000
```

Comment:
this mock repeat itself in different tests, we can use a fixture

```
mock = Mock()
mock.displaylimit = 100
mock.autolimit = 100000
```

Reply:
done